mypyc/test-data/run-strings.test

```diff
@@ -149,6 +149,14 @@ def test_str_to_bool() -> None:
         assert is_true(x)
         assert not is_false(x)
 
+def test_str_join() -> None:
+    var = 'mypyc'
```

Comment:
Add comment describing that these mimic the AST generated for f-strings, but shouldn't trigger the specialization as they are slightly different. This way it's less likely that somebody will modify these in the future in a way that makes these no longer useful.